homeassistant/components/tractive/__init__.py

```diff
@@ -70,12 +87,30 @@ async def cancel_listen_task(_):
     return True
 
 
+async def _generate_trackables(client, trackable):
+    """Generate trackables."""
+    trackable = await trackable.details()
+    tracker = client.tracker(trackable["device_id"])
+
+    tracker_details, hw_info, pos_report = await asyncio.gather(
+        tracker.details(), tracker.hw_info(), tracker.pos_report()
+    )
+
+    return {
```

Comment:
Make this a class/namedtuple/dataclass instead of passing dicts + key constants around.